pkg/dataset/config_test.go

```diff
@@ -0,0 +1,74 @@
+package dataset
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewConfig(t *testing.T) {
+	testCases := []struct {
+		name     string
+		filename string
+		input    string
+		cfg      Config
+		err      string
+	}{
+		{
+			name:     "missing config file",
+			filename: "non-existent",
+			err:      "open non-existent: no such file or directory",
+		},
+		{
+			name:     "invalid config yaml",
+			filename: "config.yaml",
+			input:    "invalid",
+			err:      "yaml: unmarshal errors:\n  line 1: cannot unmarshal !!str `invalid` into dataset.Config",
+		},
+		{
+			name:     "happy path",
+			filename: "config.yaml",
+			input:    "metrics:\n  default_chunk_interval: 3h",
+			cfg:      Config{Metrics: Metrics{ChunkInterval: 3 * time.Hour}},
+		},
+	}
+
+	for _, c := range testCases {
+		t.Run(c.name, func(t *testing.T) {
+			if c.input != "" {
+				f, err := os.CreateTemp(os.TempDir(), c.filename)
+				if err != nil {
+					t.Fatal("unable to create temp file", err)
+				}
+				c.filename = f.Name()
+
+				n, err := f.Write([]byte(c.input))
+
+				if err != nil {
+					t.Fatal("unable to write temp file", err)
+				}
+
+				if n != len(c.input) {
+					t.Fatal("unable to write complete temp file")
+				}
+				f.Close()
+				if err != nil {
+					t.Fatal("unable to write temp file", err)
+				}
+
+				defer func() { os.Remove(c.filename) }()
```

Comment:
Due to iteration and the fact that this will not evaluate immediately the defer would only remove the last iterated value. I think we should either call function directly: `defer os.Remove(c.filename)` (which evaluates immediately) or lock the value with the function by passing the parameter: `defer func(filename string){ os.Remove(filename}(c.filename)`